### PR TITLE
EVG-14490: Refactor fetching Jira ticket information

### PR DESCRIPTION
**Pull request by malikchaya2**
Refactor fetching Jira ticket information so that an error doesn't break the whole Task page.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -2690,7 +2690,11 @@
     "@leafygreen-ui/popover" "^7.1.0"
     react-transition-group "^4.4.1"
 
+<<<<<<< HEAD
```
Comment (khelif96) [inline yarn.lock:2693]: It looks like you accidentally committed some merge markers

File: src/pages/task/taskTabs/buildBaronAndAnnotations/BuildBaronContent.tsx [hunk 2]
```diff
@@ -48,7 +48,7 @@ export const BuildBaronContent: React.FC<BuildBaronCoreProps> = ({
       {loading && <Skeleton active title={false} paragraph={{ rows: 4 }} />}
       {annotation?.webhookConfigured ? (
         <CustomCreatedTickets
-          tickets={annotation?.createdIssues}
+          // tickets={annotation?.createdIssues}
```
Comment (khelif96) [inline src/pages/task/taskTabs/buildBaronAndAnnotations/BuildBaronContent.tsx:51]: Accidentally commented out.

File: src/gql/queries/get-jira-custom-created-issues.graphql [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#import "../fragments/jiraTicket.graphql"
+
+query GetCustomCreatedIssues($taskId: String!, $execution: Int) {
+  task(taskId: $taskId, execution: $execution) {
+    annotation {
```
Comment (khelif96) [inline src/gql/queries/get-jira-custom-created-issues.graphql:5]: task is missing an id and execution

Comment (khelif96) [inline src/gql/queries/get-jira-custom-created-issues.graphql:5]:
```suggestion
  task(taskId: $taskId, execution: $execution) {
    id
    execution
    annotation {
```

Comment (khelif96) [reply src/gql/queries/get-jira-custom-created-issues.graphql:5]: Apollo uses these fields to uniquely identify the returned task, So they always need to exist in a task response.

File: src/gql/queries/get-jira-issues.graphql [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#import "../fragments/jiraTicket.graphql"
+
+query GetIssues($taskId: String!, $execution: Int) {
+  task(taskId: $taskId, execution: $execution) {
```
Comment (khelif96) [inline src/gql/queries/get-jira-issues.graphql:4]: Same comment as above

File: src/gql/queries/get-jira-suspected-issues.graphql [hunk 5]
```diff
@@ -0,0 +1,20 @@
+#import "../fragments/jiraTicket.graphql"
+
+query GetSuspectedIssues($taskId: String!, $execution: Int) {
+  task(taskId: $taskId, execution: $execution) {
```
Comment (khelif96) [inline src/gql/queries/get-jira-suspected-issues.graphql:4]: here to

**Review by khelif96 — COMMENTED** [review]
Great job on the refactor!

File: src/pages/task/taskTabs/buildBaronAndAnnotations/AnnotationTickets.tsx [hunk 6]
```diff
@@ -19,7 +28,7 @@ interface Props {
   setSelectedRowKey: React.Dispatch<React.SetStateAction<string>>;
 }
 
-export const AnnotationTickets: React.FC<Props> = ({
+export const AnnotationTickets: React.FC<AnnotationTicketsProps> = ({
```
Comment (khelif96) [inline src/pages/task/taskTabs/buildBaronAndAnnotations/AnnotationTickets.tsx:31]: This doesn't need to be exported

File: src/pages/task/taskTabs/buildBaronAndAnnotations/BuildBaronContent.tsx (hunk 2 above)
Comment (malikchaya2) [reply src/pages/task/taskTabs/buildBaronAndAnnotations/BuildBaronContent.tsx:51]: ah, I need to remove that. thanks!

File: yarn.lock (hunk 1 above)
Comment (malikchaya2) [reply yarn.lock:2693]: strange, thanks!

File: yarn.lock [hunk 7]
```diff
@@ -18505,4 +18505,4 @@ yocto-queue@^0.1.0:
 zen-observable@^0.8.14:
   version "0.8.15"
   resolved "https://registry.yarnpkg.com/zen-observable/-/zen-observable-0.8.15.tgz#96415c512d8e3ffd920afd3889604e30b9eaac15"
-  integrity sha512-PQ2PC7R9rslx84ndNBZB/Dkv8V8fZEpk83RLgXtYd0fwUgEjseMn1Dgajh2x6S8QbZAFa9p2qVCEuYZNgve0dQ==
+  integrity sha512-PQ2PC7R9rslx84ndNBZB/Dkv8V8fZEpk83RLgXtYd0fwUgEjseMn1Dgajh2x6S8QbZAFa9p2qVCEuYZNgve0dQ==
```
Comment (khelif96) [inline yarn.lock:18508]: Can you remove this change. It isn't related to your pr your editor might have changed it by accident.

File: src/pages/task/taskTabs/buildBaronAndAnnotations/AnnotationTickets.tsx [hunk 8]
```diff
@@ -91,6 +100,94 @@ export const AnnotationTickets: React.FC<Props> = ({
...
+    onError: (err) => {
+      dispatchToast.error(
+        `There was an error loading the ticket information from Jira: ${err.message}`
+      );
+    },
+  });
+
+  return (
+    <AnnotationTickets
+      tickets={data?.task?.annotation?.suspectedIssues}
```
Comment (khelif96) [inline src/pages/task/taskTabs/buildBaronAndAnnotations/AnnotationTickets.tsx:180]: Can you extract this out into a variable

**Review by khelif96 — COMMENTED** [review]
Everything looks great just 2 more comments

**Pull request merged**